pandas/io/common.py

```diff
@@ -394,13 +394,15 @@ def _get_handle(path_or_buf, mode, encoding=None, compression=None,
     elif is_path:
         if compat.PY2:
             # Python 2
+            if mode == 'w':
+                mode = 'wb'
             f = open(path_or_buf, mode)
         elif encoding:
             # Python 3 and encoding
-            f = open(path_or_buf, mode, encoding=encoding)
+            f = open(path_or_buf, mode, encoding=encoding, newline='\n')
```

Comment:
@deflatSOCO Is there a specific reason for not making newline a function parameter? I would suggest that you make _newline_ a `_get_handle` parameter and default it to *None* then let `open` handle it from there.
Then update `formats.csvs.CSVFormatter.save` to pass _self.line_terminator_ to the function